content/blog/rn-bridging-native-modules/index.md

```diff
@@ -0,0 +1,339 @@
+
+```m
+// MyWebView.m
+#import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
+#import <React/RCTViewManager.h>
+
+@interface RCT_EXTERN_MODULE(MyWebViewManager, RCTViewManager)
+@end
+```
+
+That is all we need to bridge our very simple native WebView to React Native. Now, let's switch to our React Native code to try it out!
+
+## React Native
+
+```js
+import React from "react"
+import { View, requireNativeComponent, NativeModules } from "react-native"
+import { WebView } from "react-native-webview"
+
+const MyWebView = requireNativeComponent("MyWebView")
+const { MyWebViewManager } = NativeModules
+
+const MyRNWebView = () => {
+    return (
+        <View style={{ flex: 1 }}>
+            <WebView
+                nativeConfig={{
+                    component: MyWebView,
+                    viewManager: MyWebViewManager,
+                }}
+            />
+        </View>
+    )
+}
+```
+
+Here we use the regular `WebView` component from `react-native-webview` but we pass our own native components as the `component` & `viewManager` in the `nativeConfig`, as per [React Native's docs](https://facebook.github.io/react-native/docs/webview#nativeconfig).
+
+![WebView in action example](./rn-webview-example.png)
+
+# Communication between React Native and Swift
+
+So far we have made our native WebView in Swift and we've managed to bridge it to React Native and view it. Great! But at this stage, it's rather limited in its usage. It can only view a single predefined URL, and it doesn't offer any information back to React Native either.  
+
+We can fix this by letting the WebView accept props from React Native and set up events from the WebView that React Native can intercept!
+
+## Passing data to Swift class as props
+
+While staying on the React Native side, let's add the property `props` to our WebView's `nativeConfig`, containing the URL:
+
+```js
+<WebView
+    nativeConfig={{
+        component: MyWebView,
+        props: {
+            webViewURL: "https://beanloop.se",
+        },
+        viewManager: MyWebViewManager,
+    }}
+/>
+```
+
+We then need to update our **MyWebView.m** file to expose the WebView's `webViewURL` variable:
+
+*Note: The same name for the class variable must also be used for the prop, in this case, `webViewURL`.*
+
+```m
+// MyWebView.m
+#import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
+#import <React/RCTViewManager.h>
+
+@interface RCT_EXTERN_MODULE(MyWebViewManager, RCTViewManager)
+
+RCT_EXPORT_VIEW_PROPERTY(webViewURL, NSString)
+
+@end
+```
+
+Finally, in our Swift class, we need to intercept the prop and handle it accordingly. Leave the Manager class alone but update the `MyWebView` class:
+
+```swift
+// MyWebView.swift
+
+...
+
+@objc(MyWebView)
+public class MyWebView: UIView, WKNavigationDelegate, WKUIDelegate {
+    var webView: WKWebView!
+    var webViewURL: NSString?
```

Comment:
I would like to see a comment here stating that this is the one that has to have the same name as the one you specified inside the React-Native code!